pkg/scheduler/scheduler.go

```diff
@@ -850,37 +857,32 @@ func (s *Scheduler) getTypeFromClusterCResourceBindings(name string) ScheduleTyp
 		return FirstSchedule
 	}
 
-	policyName := util.GetLabelValue(binding.Labels, policyv1alpha1.ClusterPropagationPolicyLabel)
-
-	policy, err := s.clusterPolicyLister.Get(policyName)
-	if err != nil {
-		return Unknown
-	}
-	placement, err := json.Marshal(policy.Spec.Placement)
+	appliedPlacement := util.GetLabelValue(binding.Annotations, util.PolicyPlacementAnnotation)
+	policyPlacement, policyPlacementStr, err := s.getClusterPlacement(binding)
 	if err != nil {
-		klog.Errorf("Failed to marshal placement of propagationPolicy %s/%s, error: %v", policy.Namespace, policy.Name, err)
 		return Unknown
 	}
-	policyPlacementStr := string(placement)
-
-	appliedPlacement := util.GetLabelValue(binding.Annotations, util.PolicyPlacementAnnotation)
 
 	if policyPlacementStr != appliedPlacement {
 		return ReconcileSchedule
 	}
 
-	if policy.Spec.Placement.ReplicaScheduling != nil && util.IsBindingReplicasChanged(&binding.Spec, policy.Spec.Placement.ReplicaScheduling) {
+	if policyPlacement.ReplicaScheduling != nil && util.IsBindingReplicasChanged(&binding.Spec, policyPlacement.ReplicaScheduling) {
 		return ScaleSchedule
 	}
 
+	return s.verifyClustersStatus(binding.Spec.Clusters)
+}
+
+func (s *Scheduler) verifyClustersStatus(tc []workv1alpha1.TargetCluster) ScheduleType {
```

Comment:
This method should focus on checking if there is a `not-ready` cluster in the cluster list.(should not couple scheduling logic)
So, I suggest the method signature would be :
```
func AllClustersInReadyState(tcs []workv1alpha1.TargetCluster) bool 
```

Reply:
since current func calls `s.schedulerCache.Snapshot().GetClusters()`, as you advised, two variables are needed, right? IMO, I think it makes sense as a method of `Scheduler`. Listening  to your advice. by the way, tc should be tcs sounds good.

Reply:
> since current func calls s.schedulerCache.Snapshot().GetClusters(), as you advised, two variables are needed, right?

Sorry, I didn't mean to abstract a common function. How about this?
```golang
func (s *Scheduler) AllClustersInReadyState(tcs []workv1alpha1.TargetCluster) bool 
```

Reply:
lgtm.

Reply:
done. :)